src/taipy/gui_core/GuiCoreLib.py

```diff
@@ -137,18 +181,20 @@ class GuiCore(ElementLibrary):
                 "show_primary_flag": ElementProperty(PropertyType.dynamic_boolean, True),
                 "value": ElementProperty(PropertyType.lov_value),
                 "on_change": ElementProperty(PropertyType.function),
-                "scenario": ElementProperty(GuiCoreScenarioAdapter),
             },
             inner_properties={
                 "scenarios": ElementProperty(PropertyType.lov, f"{{{__CTX_VAR_NAME}.get_scenarios()}}"),
-                "on_scenario_create": ElementProperty(
-                    PropertyType.function, f"{{{__CTX_VAR_NAME}.create_new_scenario}}"
-                ),
+                "on_scenario_crud": ElementProperty(PropertyType.function, f"{{{__CTX_VAR_NAME}.crud_scenario}}"),
                 "configs": ElementProperty(PropertyType.react, f"{{{__CTX_VAR_NAME}.get_scenario_configs()}}"),
                 "core_changed": ElementProperty(PropertyType.broadcast, GuiCoreContext._CORE_CHANGED_NAME),
                 "error": ElementProperty(PropertyType.react, f"{{{GuiCoreContext._ERROR_VAR}}}"),
                 "type": ElementProperty(PropertyType.inner, Scenario),
                 "adapter": ElementProperty(PropertyType.inner, GuiCoreContext.scenario_adapter),
+                "scenario_edit": ElementProperty(
+                    GuiCoreScenarioAdapter,
+                    f"{{{__CTX_VAR_NAME}.get_scenario_by_id({GuiCoreContext._SCENARIO_SELECTOR_ID_VAR})}}",
```

Comment:
Any problem if scenario was deleted at this point? -> "None"

Reply:
get_scenario_by_id returns None but the core should have trigger a tree refresh...